lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -114,6 +114,11 @@ static LogicalResult customTypePrinter(Type type, AsmPrinter &os) {
           os << ", " << layer;
         os << '>';
       })
+      .Case<LHSType>([&](LHSType lhstype) {
+        os << "lhs<";
+        printNestedType(lhstype.getType(), os);
```

Comment:
Please add printing/parsing test snippets for this?

As-is I think it's allowed in a wire, but probably that's a short-term thing (?).

Anyway if there's an IR snippet that uses these that would be great -- but adding it later sounds good if that doesn't really make sense until there are other bits.

Reply:
Yea, wire needs to handle its type with it's constraints/verifier.  I've narrowed down the places which can go wrong by moving LHSType to FIRRTLType from FIRRTLBaseType.  Doesn't fix all issues, but does many.